File: src/com/leozhi/topic136/Solution03.java
package com.leozhi.topic136;

/**
 * @author leozhi
 * 位运算
 * 通过
 * 1ms
 */
public class Solution03 {
    public int singleNumber(int[] nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }
}

File: src/com/leozhi/topic560/Solution03.kt
package com.leozhi.topic560

/**
 * @author leozhi
 * 前缀和 + 哈希表优化
 * 通过
 * 340ms
 */
class Solution03 {
    fun subarraySum(nums: IntArray, k: Int): Int {
        var preSum = 0
        val map = HashMap<Int, Int>()
        map[0] = 1
        var res = 0
        for (num in nums) {
            preSum += num
            if (map.containsKey(preSum - k)) {
                res += map[preSum - k]!!
            }
            map[preSum] = map.getOrDefault(preSum, 0) + 1
        }
        return res
    }
}